client/src/worldgen.rs

```diff
@@ -155,10 +155,89 @@ pub fn voxels(graph: &mut DualGraph, node: NodeId, cube: Vertex) -> VoxelData {
                 }
             }
         }
+
+        // Planting trees on dirt. Trees consist of a block of wood and a block of leaves.
+        // The leaf block is on the opposite face of the wood block as the dirt block.
+        let loc: usize = 2;
+        let voxel_of_interest_index =
+            index(absolute_subchunk_coords(loc, loc, loc, subchunk_offset));
+        let neighbor_data = voxel_neighbors(
+            na::Vector3::new(loc, loc, loc),
+            subchunk_offset,
+            &mut voxels,
+        );
+
+        let num_void_neighbors = neighbor_data
+            .iter()
+            .filter(|n| n.material == Material::Void)
+            .count();
+
+        // Only plant a tree if there is exactly one adjacent dirt block.
+        if num_void_neighbors == 5 {
+            for i in neighbor_data.iter() {
+                if i.material == Material::Dirt {
+                    voxels.data_mut()[voxel_of_interest_index] = Material::Wood;
+                    let leaf_location = index(i.abs_coords_opposing);
+                    voxels.data_mut()[leaf_location] = Material::Leaves;
+                }
+            }
+        }
     }
+
     voxels
 }
 
+pub struct NeighborData {
+    #[allow(dead_code)]
+    abs_coords: na::Vector3<f64>,
+    abs_coords_opposing: na::Vector3<f64>,
```

Comment:
Is `f64` justified here? Also, let's avoid adding data that we don't actually need yet; we can always put it in later, but it's harder to go back after the fact and determine what's unnecessary and can be removed.

Reply:
I will remove the unused abs_coords field.
I believe f64 is justified for the abs_coords_opposing vector because I populate it using absolute_subchunk_coords() which returns a Vector3 of f64 values and use it in the index() function which accepts a Vector3 of f64 values. I don't want to change existing functions without discussing it first.

Reply:
Fair enough.